core/dbt/events/types.py

```diff
@@ -994,6 +997,649 @@ def cli_msg(self) -> str:
+        return stats_line.format(**self.stats)
+
+
+@dataclass
+class RunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return f"  {self.msg}"
+
+
+@dataclass
+class RunResultErrorNoMessage(ErrorLevel, CliEventABC):
+    status: str
+
+    def cli_msg(self) -> str:
+        return f"  Status: {self.status}"
+
+
+@dataclass
+class SQLCompiledPath(InfoLevel, CliEventABC):
+    path: str
+
+    def cli_msg(self) -> str:
+        return f"  compiled SQL at {self.path}"
+
+
+@dataclass
+class CheckNodeTestFailure(InfoLevel, CliEventABC):
+    relation_name: str
+
+    def cli_msg(self) -> str:
+        msg = f"select * from {self.relation_name}"
+        border = '-' * len(msg)
+        return f"  See test failures:\n  {border}\n  {msg}\n  {border}"
+
+
+@dataclass
+class FirstRunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return ui.yellow(self.msg)
+
+
+@dataclass
+class AfterFirstRunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return self.msg
+
+
+@dataclass
+class EndOfRunSummary(InfoLevel, CliEventABC):
+    num_errors: int
+    num_warnings: int
+    keyboard_interrupt: bool = False
+
+    def cli_msg(self) -> str:
+        error_plural = utils.pluralize(self.num_errors, 'error')
+        warn_plural = utils.pluralize(self.num_warnings, 'warning')
+        if self.keyboard_interrupt:
+            message = ui.yellow('Exited because of keyboard interrupt.')
+        elif self.num_errors > 0:
+            message = ui.red("Completed with {} and {}:".format(
+                error_plural, warn_plural))
+        elif self.num_warnings > 0:
+            message = ui.yellow('Completed with {}:'.format(warn_plural))
+        else:
+            message = ui.green('Completed successfully')
+        return message
+
+
+def format_fancy_output_line(
+        msg: str, status: str, index: Optional[int],
+        total: Optional[int], execution_time: Optional[float] = None,
+        truncate: bool = False
+) -> str:
+    if index is None or total is None:
+        progress = ''
+    else:
+        progress = '{} of {} '.format(index, total)
+    # TODO: remove this formatting once we rip out all the old logger
+    prefix = "{timestamp} | {progress}{message}".format(
+        timestamp=get_timestamp(),
+        progress=progress,
+        message=msg)
+
+    truncate_width = ui.printer_width() - 3
+    justified = prefix.ljust(ui.printer_width(), ".")
+    if truncate and len(justified) > truncate_width:
+        justified = justified[:truncate_width] + '...'
+
+    if execution_time is None:
+        status_time = ""
+    else:
+        status_time = " in {execution_time:0.2f}s".format(
+            execution_time=execution_time)
+
+    output = "{justified} [{status}{status_time}]".format(
+        justified=justified, status=status, status_time=status_time)
+
+    return output
```

Comment:
I'm thinking we should move this into `functions.py` so that everything here is a type definition. Does that sound reasonable to you?

Reply:
That creates a circular dependancy.  You're right that it doesn't belong in types though.  Maybe a new module in /events?  Then all the formatting (coloring etc) can end up there eventually too.  `format.py`?

Reply:
sure! I dig it. Let's make it happen in this PR just because we can.